### ISSUE TITLE
Cancel external compactions when a user compaction is canceled

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#2028]
Internal compaction use [this code][1] to determine if they should keep running a user compaction.  Each user compaction has a 1 up per table id.  There is also a per table cancel id in ZK, if the cancelId >= userCompactionId then the user compaction was canceled. 

[1]:https://github.com/apache/accumulo/blob/42de4efab84b4dc0e4eda23c6d348be744941c30/server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java#L738-L757

**Comment by dlmarion** [comment]
Wrote ExternalCompactionIT.testUserCompactionCancellation in commit b21af9fea21bff53f3779c057ddc5bc16c730ac9

**Comment by keith-turner** [comment]
> Wrote ExternalCompactionIT.testUserCompactionCancellation in commit b21af9f

@dlmarion  the test look great.  I tweaked them a bit in 2c6095edd1592037b00a178e1e68b6b49001577c

When running them I saw the following error.

```
[INFO] Running org.apache.accumulo.test.ExternalCompactionIT
[ERROR] Tests run: 7, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 602.636 s <<< FAILURE! - in org.apache.accumulo.test.ExternalCompactionIT
[ERROR] org.apache.accumulo.test.ExternalCompactionIT.testSplitDuringExternalCompaction  Time elapsed: 49.06 s  <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<2>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at org.apache.accumulo.test.ExternalCompactionIT.testSplitDuringExternalCompaction(ExternalCompactionIT.java:227)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Line 227 is `assertEquals(1, metrics.getStarted());` any idea why that was off?

**Comment by dlmarion** [comment]
My guess is that it split the table and then another compaction kicked off. There were other places in the code where I changed `assertEquals(1, metrics.getStarted())` to `assertTrue(metrics.getStarted() > 0)`. I didn't necessarily care that it the value was `1`, just that it was non-zero.

**Comment by dlmarion** [comment]
Fixed in f8b906f, I think we can close this.
